TWLight/applications/models.py

```diff
@@ -25,6 +25,14 @@ def get_queryset(self):
         return super(ValidApplicationsManager, self).get_queryset(
             ).exclude(status=Application.INVALID)
 
+
+def get_latest_app(partner, editor):
```

Comment:
This can be directly on the Authorization model so that you don't need to pass the partner and editor info in. See e.g. the functions at https://github.com/WikipediaLibrary/TWLight/blob/master/TWLight/users/models.py#L220

Reply:
Then you wouldn't need to do https://github.com/WikipediaLibrary/TWLight/blob/5c5797f4b9018cb93ecbda2cb59d9cb810bf387a/TWLight/users/views.py#L620 because you could access that data directly, without needing to loop through each authorization object and adding new data. In fact, the same could be said for expiring soon and expired. If those were functions on the Authorization model we could potentially do away with needing those loops at all?

Reply:
importing Application model in `users.model` results in all sorts of errors. Not sure how to proceed.

Reply:
is this what is called a 'circular import'? Should I try to import from within the function?

Reply:
That issue is avoidable in Django, see https://github.com/WikipediaLibrary/TWLight/blob/28a9ce3f77406c8613596badc3a22072d20018f9/TWLight/resources/models.py#L578.

Reply:
sorry, but I don't fully understand what you are trying to ask of me. Establish Many-to-one relationship from Application to Authorization and then tweak the code to set `Authorization.pk` for every application that is being created? If you had something else in your mind, please explain.

Reply:
Ah, sorry for the confusion, I'm probably using the wrong wording. I'm referring to adding model methods (see https://django-book.readthedocs.io/en/latest/chapter10.html#model-methods) to the Authorization model so this functionality lives more centrally.

On the user model, for example, we have a `get_absolute_url` method (https://github.com/WikipediaLibrary/TWLight/blob/master/TWLight/users/models.py#L374) which can be called on any user object. We could have `get_latest_app` be a method on the Authorization model so that by default you can always do `authorization_object.get_latest_app()` to find the application associated with an Authorization object.

See the bottom of https://gist.github.com/Samwalton9/aee729e6f4faa8347e0ca99f2f40d6e2 for an (untested) example of what I mean. With `get_latest_app` as a model method directly, you then wouldn't need to loop through all your applications setting that information here, because the function would already be accessible to you directly on the model.

Does that clarify?

Reply:
Yep, that clarifies things. Thanks!